pandas/tests/indexes/multi/test_constructor.py

```diff
@@ -473,6 +473,28 @@ def test_from_product_datetimeindex():
     tm.assert_numpy_array_equal(mi.values, etalon)
 
 
+@pytest.mark.parametrize(
+    "a, b, expected_names",
+    [
+        (
+            pd.Series([1, 2, 3], name="foo"),
+            pd.Series(["a", "b"], name="bar"),
+            ["foo", "bar"],
+        ),
+        (pd.Series([1, 2, 3], name="foo"), ["a", "b"], ["foo", None]),
+        ([1, 2, 3], ["a", "b"], None),
+    ],
```

Comment:
Can you add a test case for `pd.Series([1, 2, 3]), pd.Series(['a' 'b'])`, where both the names are None, to ensure we hit that condition?

Reply:
Added this test case